ledger/eval.go

```diff
@@ -385,7 +386,7 @@ func (eval *BlockEvaluator) transactionGroup(txgroup []transactions.SignedTxnWit
 	for _, txad := range txgroup {
 		var txib transactions.SignedTxnInBlock
 
-		err := eval.transaction(txad.SignedTxn, txad.ApplyData, cow, &txib)
+		err := eval.transaction(txad.SignedTxn, txad.ApplyData, txgroup, cow, &txib)
```

Comment:
We now have a bug in our transaction group code (that previously didn't matter but does now):
If someone submits a list of transactions all with zero "group" field to our transaction pool claiming it's all one transaction group, we'll give it to our transaction pool's block evaluator as a single group, and the evaluator will accept it. But when other people see the block, they'll (correctly) interpret each of these transactions as a size-1 group of its own. Previously, this made no difference. But now, a logicsig transaction's validity may depend on whether a transaction is part of a group.
In particular, suppose one of these transactions has logic that rejects if the transaction is not part of a group of size 2 or larger. The node this group is submitted to will propose an invalid block but consider it valid, but everyone else will reject the block.
Probably the fix is to have the `transactionGroup()` function return an error at the beginning if the length of the transaction group is >= 1 but the group field is zero

Reply:
I posted a fix to this (pretty sure). Added a check a few lines below that rejects something submitted as a txn group that contains txns not tagged as part of a group